airflow/providers/common/sql/hooks/sql.py

```
@@ -210,6 +210,17 @@ def connection(self) -> Connection:
             self._connection = self.get_connection(self.get_conn_id())
         return self._connection
 
+    @connection.setter
+    def connection(self, value: Any) -> None:
+        # This dummpy setter is for backward compatibility and should not be used.
+        # Since the introduction of connection property, the providers listed below
+        # breaks due to assigning value to self.connection
+        #
+        # apache-airflow-providers-mysql<5.7.1
+        # apache-airflow-providers-elasticsearch<5.5.1
+        # apache-airflow-providers-postgres<5.13.0
+        pass
```

So assigning to this property now silently does nothing… how does this work for those providers?

---

they still have `self.connection = ...`

---

+1